index.js

```diff
@@ -1 +1,113 @@
 // implement your API here
+const express = require('express');
+const cors = require('cors');
+const db = require('./data/db');
+const server = express();
+server.use(express.json());
+server.use(cors());
+
+// Delivers 'server running' response:
+server.get('/', (req, res) => {
+  res.send('Server running...');
+});
+
+// Fetches users array:
+server.get('/api/users', (req, res) => {
+  db.find()
+    .then(users => {
+      if (users) {
+        res.status(200).json(users);
+      } else {
+        res.status(404).json({ message: 'no users to return' });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The users information could not be retrieved.' });
+    });
+});
+
+// Fetches a user:
+server.get('/api/users/:id', (req, res) => {
+  const id = req.params.id;
+  db.findById(id)
+    .then(user => {
+      if (user) {
+        res.status(200).json(user);
+      } else {
+        res
+          .status(404)
+          .json({ message: 'The user with the specified ID does not exist.' });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'The user information could not be retrieved.' });
+    });
+});
+
+// Adds a user:
+server.post('/api/users', (req, res) => {
+  const user = req.body;
+  if (user.name && user.bio) {
+    db.insert(user)
+      .then(user => res.status(201).json(user))
+      .catch(err => res.status(500).json({ message: 'error adding user' }));
+  } else {
+    res
+      .status(400)
+      .json({ errorMessage: 'Please provide name and bio for the user.' });
+  }
+});
+
+// Deletes a user:
+server.delete('/api/users/:id', (req, res) => {
+  const id = req.params.id;
+  db.remove(id)
+    .then(user => {
+      if (user === 1) {
+        // Sends general 'success' code, then ends the transmission (I forgot the proper name) since there is no need to send any other information.
```

Comment:
this can be focused a bit by remembering that a delete request to this database will return the number of deleted users, so instead of saying `user` being passed in to the callback, I'd say `num`, and then since that number will only ever be 0 or a positive integer you could simply check the truthiness value of (zero is falsey) like `if (num) {do stuff}`